src/components/MovieList/MovieList.jsx

```diff
@@ -0,0 +1,111 @@
+import { Button, CardContainer } from 'components/MovieList/styles';
+import React, { useState } from 'react';
+
+import MovieCard from 'components/MovieCard/MovieCard';
+import PropTypes from 'prop-types';
+import blurImage from 'assets/image.jpg';
+import dayjs from 'dayjs';
+import { fetchDataFromApiList } from 'utils/function';
+
+// eslint-disable-next-line import/no-mutable-exports
+export let pageNumber = 1;
+/**
+ * MovieList Component
+ * Displays a list of movies in a paginated manner and provides an option to load more movies.
+ *
+ * @param {Array} movies - The array of movies to display.
+ * @param {Function} setMovies - Function to set the movies in the parent component.
+ * @param {Function} setParams - Function to set parameters in the parent component.
+ * @param {Object} params - Parameters for the movie list.
+ * @returns {JSX.Element} JSX element representing the movie list.
+ */
+export default function MovieList({ movies, setMovies, setParams, params }) {
+  const [isMenuOpen, setIsMenuOpen] = useState('');
+
+  /**
+   * Handles toggling the menu open/close state for a specific movie.
+   * If the movie's menu is currently open, it closes it. If closed, it opens it.
+   *
+   * @param {string} movieId - The unique identifier of the movie.
+   */
+  const menuOpenHandler = (movieId) => {
+    if (isMenuOpen === movieId) {
+      setIsMenuOpen('');
+    } else {
+      setIsMenuOpen(movieId);
+    }
+  };
+
+  /**
+   * Handles loading more movies by making an API call to fetch additional movie data.
+   */
+  const loadMoreHandler = () => {
+    // eslint-disable-next-line no-plusplus
```

Comment:
why use eslint-disable here? normally we want our code to follow some standards and rules to be consistent and have it better organized. if we have a rule specified to not use the `plusplus` operator, then we should use a different approach to increment the variable. `pageNumber += 1` for example. 

However, if you are the one setting up the eslint rules and need to have your own set of configs and rules, then you can enable the plusplus operator by editing the eslintrc.js to disable the no-plusplus rule as follow: 
```
    rules: {
    'no-plusplus': 'off',
  },
```